### PR TITLE
Rx Refactor OutdatedAppViewModel

### PR DESCRIPTION
**Pull request by khoi**
## What's this?
Rx Refactor OutdatedAppViewModel

### Relationships
Part of #3352

## Why do we want this?
To make our lives easier.

## How is it done?
By rx-ifying stuff

### Side effects
None.

## Review hints
The app's having bugs displaying this screen #3484, #3483.

So if you want to test the functionality of this, probably call this in `MainViewModel` somewhere

```
navigationService.Navigate<OutdatedAppViewModel>();
``` 

## :squid: Permissions
🦑

### CODE REVIEW

File: Toggl.Foundation.MvvmCross/ViewModels/OutdatedAppViewModel.cs [hunk 1]
```diff
@@ -13,28 +17,41 @@ public sealed class OutdatedAppViewModel : MvxViewModel
...
-            OpenWebsiteCommand = new MvxCommand(openWebsite);
+            UpdateAppAction = new UIAction(updateApp);
+            OpenWebsiteAction = new UIAction(openWebsite);
         }
 
-        private void openWebsite()
+        private IObservable<Unit> openWebsite()
         {
-            browserService.OpenUrl(togglWebsiteUrl);
+            return Observable.Create<Unit>(observer =>
```
Comment (simonrozsival) [inline Toggl.Foundation.MvvmCross/ViewModels/OutdatedAppViewModel.cs]: Let's use an arrow function here to minimize the amount of boilerplate.

File: Toggl.Foundation.MvvmCross/ViewModels/OutdatedAppViewModel.cs [hunk 2]
```diff
@@ -13,28 +17,41 @@ public sealed class OutdatedAppViewModel : MvxViewModel
...
+            {
+
+                browserService.OpenUrl(togglWebsiteUrl);
+                observer.OnNext(Unit.Default);
```
Comment (simonrozsival) [inline Toggl.Foundation.MvvmCross/ViewModels/OutdatedAppViewModel.cs]: Use the `.CompleteWith(Unit.Default)` extension method to remove one extra line!

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/OutdatedAppViewModelTests.cs [hunk 3]
```diff
@@ -28,12 +28,12 @@ public void ThrowsIfTheArgumentIsNull()
...
         }
 
-        public sealed class TheUpdateAppCommand : OutdatedAppViewModelTest
+        public sealed class TheUpdateAppAction : OutdatedAppViewModelTest
         {
             [Fact, LogIfTooSlow]
             public void CallsTheOpenStoreMethodOfTheBrowserService()
             {
-                ViewModel.UpdateAppCommand.Execute();
+                ViewModel.UpdateAppAction.Execute();
```
Comment (kornelijepetak) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/OutdatedAppViewModelTests.cs]: Execute is an async action, is it not?
I think you should await both this and the next call.

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/OutdatedAppViewModelTests.cs [hunk 4]
```diff
@@ -42,11 +42,11 @@ public void CallsTheOpenStoreMethodOfTheBrowserService()
         public sealed class TheOpenWebsiteCommand : OutdatedAppViewModelTest
         {
             [Fact, LogIfTooSlow]
-            public void TogglesTheCurrentValueOfTheToggleUseTwentyFourHourClockProperty()
+            public void CallsTheOpenWebsiteMethodOfTheBrowserService()
             {
                 const string togglWebsiteUrl = "https://toggl.com";
 
-                ViewModel.OpenWebsiteCommand.Execute();
+                ViewModel.OpenWebsiteAction.Execute();
```
Comment (kornelijepetak) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/OutdatedAppViewModelTests.cs]: The same thing about `await`.
Check the [AboutViewModel](https://goo.gl/4vHCtR).

**Pull request merged**